git-buildkite

```diff
@@ -176,14 +184,20 @@ if [ -z "$BUILDKITE_ORGANIZATION" -o -z "$BUILDKITE_PIPELINE" ]; then
 fi
 
 # Parse flags before parsing arguments
-let i=1
-while [[ $i -le $# ]]; do
-  case "${!i}" in
+while [[ $# -gt 0 ]]; do
```

Comment:
The previous code didn't quite work here due to the usage of `shift $i`. This newer version is something I see fairly often in Bash, and serves to strip out any switches, then positional args (which should be branch/commit) start at $1 for the parts of the script below 201. Happy to take another look if I am missing something here.

Reply:
Looks good to me.

Reply:
I use this pretty commonly:

```
git buildkite main HEAD --browse
```

I think that would break in this version?